utt/privacy-wallet-service/src/PrivacyService.cpp

```diff
@@ -389,4 +395,30 @@ ::grpc::Status PrivacyWalletServiceImpl::handleGetStateRequest(
   }
   return grpc::Status::OK;
 }
+
+::grpc::Status PrivacyWalletServiceImpl::handleSetAppDataRequest(
+    ::grpc::ServerContext*,
+    const ::vmware::concord::privacy::wallet::api::v1::PrivacyWalletRequest* request,
+    ::vmware::concord::privacy::wallet::api::v1::PrivacyWalletResponse* response) {
+  auto& set_app_data_req = request->set_app_data_request();
+  storage_->setAppData(set_app_data_req.key(), set_app_data_req.value());
+  if (response) {
+    auto set_app_data_resp = response->mutable_set_app_data_response();
+    set_app_data_resp->set_succ(true);
+  }
+  return grpc::Status::OK;
+}
+
+::grpc::Status PrivacyWalletServiceImpl::handleGetAppDataRequest(
+    ::grpc::ServerContext*,
+    const ::vmware::concord::privacy::wallet::api::v1::PrivacyWalletRequest* request,
+    ::vmware::concord::privacy::wallet::api::v1::PrivacyWalletResponse* response) {
+  auto& get_app_data_req = request->get_app_data_request();
+  auto get_app_data_resp = response->mutable_get_app_data_response();
```

Comment:
Nit: move those two lines under the check for `response` as they are only used in that scope.